drip_page_parser.js

```diff
@@ -0,0 +1,174 @@
+  var $jq = window.jQuery;
+  $jq.noConflict();
+  $jq.fn.exists = function () {
+    if (this.length > 0) {
+      return this;
+    } else {
+      return false;
+    }
+  };
+
+  //define methods needed for parsing
+  var hexRegEx = new RegExp(/^#[0-9a-f]{3,6}$/i);
+  var rgb2hex = function (rgb) {
+    if (!rgb || rgb.indexOf("rgb") != 0) {
+      return rgb || "-";
+    }
+    var rgbArr = rgb.match(/^rgb\((\d+),\s*(\d+),\s*(\d+)\)$/);
+
+    if (!rgbArr) {
+      rgbArr = rgb.match(/^rgba\((\d+),\s*(\d+),\s*(\d+),\s*(\d+)\)$/);
+    }
+    function hex(x) {
+      return ("0" + parseInt(x).toString(16)).slice(-2);
+    }
+    if (rgbArr && rgbArr[4] && rgbArr[4] == "0") {
+      return "transparent";
+    } else if (rgbArr) {
+      return "#" + hex(rgbArr[1]) + hex(rgbArr[2]) + hex(rgbArr[3]);
+    } else {
+      return "ERR";
+    }
+  };
+  var getTypeSet = function (el, name) {
+    return {
+      name: name,
+      "text-colour": rgb2hex(el.css("color") || naMsg),
+      font: el.css("font-family") || naMsg,
+      "font-size": el.css("font-size") || naMsg,
+      leading: el.css("line-height") || naMsg,
+      "font-style": el.css("font-style") || naMsg,
+    };
+  };
+  var orderColourResponse = function (colourArr, prependValueArr) {
+    colourArr = (colourArr || []).sort(function (a, b) {
+      return b[1] < a[1] ? -1 : b[1] > a[1] ? 1 : 0;
+    });
+
+    $jq.each(prependValueArr, function (i, el) {
+      var from =
+        $jq.map(colourArr, function (n, j) {
+          return n[0] == el ? j : null;
+        })[0] || 0;
+      colourArr.splice(i, 0, colourArr.splice(from, 1)[0]);
+    });
+    return colourArr;
+  };
+
+  //define vars needed for parsing
+  var images = $jq(document.images);
+  var imgPaths = [];
+  var img1, img2, img3;
+
+  //select base elemts to run selectors from
+  var baseSelector =
+    $jq("[role=main]").exists() ||
+    $jq("#main").exists() ||
+    $jq("#content").exists() ||
+    $jq(document.body);
+  var h1 = baseSelector.find("h1:first").exists() || $jq("h1:first");
+  var h2 = baseSelector.find("h2:first").exists() || $jq("h2:first");
+  var h3 = baseSelector.find("h3:first").exists() || $jq("h3:first");
+  var h4 = baseSelector.find("h4:first").exists() || $jq("h4:first");
+  var h5 = baseSelector.find("h5:first").exists() || $jq("h5:first");
+  var h6 = baseSelector.find("h6:first").exists() || $jq("h6:first");
+  var p = baseSelector.find("p:first").exists() || $jq("p:first");
+  var a = baseSelector.find("a:first").exists() || $jq("a:first");
+  var body = $jq(document.body);
+  var naMsg = "N/A";
+
+  var coloursBgReturn = [],
+    coloursTextReturn = [];
+  var colourAttributes = {},
+    result = {};
+  var colour;
+  var colourOccurences = {};
+
+  //select images to return
+  if (images.length >= 3) {
+    img1 = images[parseInt(images.length / 2) - 1];
+    img2 = images[parseInt(images.length / 2)];
+    img3 = images[parseInt(images.length / 2) + 1];
```

Comment:
It's just picking the middle three images?? 😕  Good thing we found that other library